_reports/palash25.md

```diff
@@ -0,0 +1,187 @@
+      - details: >
+         Processing: Modify tests to use actual files instead of hard coded
+         tuples as file content.
+  - 15:
+      - repo: coala
+      - link: https://github.com/coala/coala/commit/3ae03bec665943aa6173047245ca929a0a754c69
+      - details: >
+         Added support for collection types like `dict` and `set` for
+         `persistent_hash`.
+---
+
+### Optimize Caching for the NextGen-Core
+
+#### Work Done
+
+1. Improved I/O mechanism for coala.
+   1. Added a class `FileFactory` to interface with files and to provide file
+      contents in different forms like string, binary data or as a tuple.
+   2. Added a `Directory` class to interface with directories and provide
+      useful information about them such as the directory path, parent
+      directory and timestamps. This will be later used to implement ignore
+      directories functionality.
+   3. Added documentation for `FileFactory` and improved the NextGen-Core
+      docs.
+
+2. Caching support.
+   1. Cached the properties of `FileFactory` which turned out to be a
+      significant performance boost in consecutive coala runs.
+   2. Added support for unordered collection types in `persistent_hash`
+      which improved the NextGen-Core's ability to hash much more
+      complex task objects (which are used in NextGen caching).
+
+3. Integrated `FileFactory` with the core.
+   1. Added line endings support for `FileFactory`.
+   2. Implemented a middleware `FileDict` that mimics a dictionary
+      and provides the bears in the old core with the actual file
+      contents instead of the `FileFactory` objects thereby maintaining
+      backwards compatibility with the old core.
+   3. Modified the tests to use actual files and the `FileFactory` objects
+      instead of hard-coded file contents.
+
+#### Challenges
+
+One of the first challenges I faced was designing the `FileFactory`.
+But with my mentors help we were able to come up with a design that
+could be used for the NextGen-Core and also maintained compatibility
+with the old core.
+
+The biggest challenge that I faced these three months for the integration
+of the aforementioned `FileFactory` with the core. Initially about thirty
+tests were failing but after discussing the idea of using a middleware
+(`FileDict`) to maintain backwards compatibilty with the old core the test
+failures were significantly reduced to just six. After a while I discovered
+that `FileFactory` was somehow removing line-endings from the file contents
+which was fixed by adding support for newlines in `FileFactory` which in turn
+led to a successful integration.
+
+#### Work to be done
+
+Two major features were left out because of time constraints:
+1. Ignore directories functionality for the NextGen-Core.
+2. Cache control flags for the NextGen-Core (which was
+   already mentioned as a stretch issue in my GSoC proposal).
+
+I have started the work on implementing the cache control flags and will
```

Comment:
remove this paragraph.

Work to be done is not about *you* doing the work; it is just stuff which is left over.

While we hope you'll continue, this work report is the end of your commitment.